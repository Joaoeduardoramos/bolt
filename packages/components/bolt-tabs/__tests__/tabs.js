// import {
//   render,
//   renderString,
//   stop as stopTwigRenderer,
// } from '@bolt/twig-renderer';
// import { fixture as html } from '@open-wc/testing-helpers';
// const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
// const { join } = require('path');
// const schema = readYamlFileSync(join(__dirname, '../tabs.schema.yml'));
// const { single, spacing } = schema.properties;

// async function renderTwig(template, data) {
//   return await render(template, data, true);
// }

// async function renderTwigString(template, data) {
//   return await renderString(template, data, true);
// }

// const timeout = 120000;

// describe('<bolt-tabs> Component', () => {
//   let page;

//   beforeEach(async () => {
//     page = await global.__BROWSER__.newPage();
//     await page.goto('http://127.0.0.1:4444/', {
//       waitUntil: 'networkidle0',
//     });
//   }, timeout);

//   afterAll(async () => {
//     await stopTwigRenderer();
//   }, timeout);

//   test('basic usage', async () => {
//     const results = await renderTwig('@bolt-components-tabs/tabs.twig', {
//       items: [
//         {
//           label: 'Tabs item 1',
//           content: 'This is the tabs content.',
//         },
//         {
//           label: 'Tabs item 2',
//           content: 'This is the tabs content.',
//         },
//         {
//           label: 'Tabs item 3',
//           content: 'This is the tabs content.',
//         },
//       ],
//     });
//     expect(results.ok).toBe(true);
//     expect(results.html).toMatchSnapshot();
//   });

//   single.enum.forEach(async singleChoice => {
//     test(`expand single items: ${singleChoice}`, async () => {
//       const results = await renderTwig('@bolt-components-tabs/tabs.twig', {
//         single: singleChoice,
//         items: [
//           {
//             label: 'Tabs item 1',
//             content: 'This is the tabs content.',
//           },
//           {
//             label: 'Tabs item 2',
//             content: 'This is the tabs content.',
//           },
//           {
//             label: 'Tabs item 3',
//             content: 'This is the tabs content.',
//           },
//         ],
//       });
//       expect(results.ok).toBe(true);
//       expect(results.html).toMatchSnapshot();
//     });
//   });

//   spacing.enum.forEach(async spacingChoice => {
//     test(`spacing: ${spacingChoice}`, async () => {
//       const results = await renderTwig('@bolt-components-tabs/tabs.twig', {
//         spacing: spacingChoice,
//         items: [
//           {
//             label: 'Tabs item 1',
//             content: 'This is the tabs content.',
//           },
//           {
//             label: 'Tabs item 2',
//             content: 'This is the tabs content.',
//           },
//           {
//             label: 'Tabs item 3',
//             content: 'This is the tabs content.',
//           },
//         ],
//       });
//       expect(results.ok).toBe(true);
//       expect(results.html).toMatchSnapshot();
//     });
//   });

//   test('Default <bolt-tabs> with Shadow DOM renders', async function() {
//     const defaultTabsShadowRoot = await page.evaluate(() => {
//       const tabs = document.createElement('bolt-tabs');
//       tabs.innerHTML = `
//         <bolt-tab-panel>
//           <bolt-text slot="label">Tabs item 1</bolt-text>
//           <bolt-text>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</bolt-text>
//         </bolt-tab-panel>
//         <bolt-tab-panel>
//           <bolt-text slot="label">Tabs item 2</bolt-text>
//           <bolt-text>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</bolt-text>
//         </bolt-tab-panel>
//         <bolt-tab-panel>
//           <bolt-text slot="label">Tabs item 3</bolt-text>
//           <bolt-text>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</bolt-text>
//         </bolt-tab-panel>`;

//       document.body.appendChild(tabs);
//       tabs.updated();

//       const child = tabs.querySelector('bolt-tab-panel');
//       child.updated();

//       return child.renderRoot.innerHTML;
//     });

//     const renderedShadowDomHTML = await html(defaultTabsShadowRoot);

//     await page.waitFor(500);
//     const image = await page.screenshot();

//     expect(image).toMatchImageSnapshot({
//       failureThreshold: '0.01',
//       failureThresholdType: 'percent',
//     });

//     // @todo: this just renders the <style> tag, same happens in button
//     // Is there any point in adding this snapshot?
//     // expect(renderedShadowDomHTML).toMatchSnapshot();
//   });

//   test('Default <bolt-tabs> w/o Shadow DOM renders', async function() {
//     const defaultTabsShadowRoot = await page.evaluate(() => {
//       const tabs = document.createElement('bolt-tabs');
//       tabs.innerHTML = `
//         <bolt-tab-panel>
//           <bolt-text slot="label">Tabs item 1</bolt-text>
//           <bolt-text>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</bolt-text>
//         </bolt-tab-panel>
//         <bolt-tab-panel>
//           <bolt-text slot="label">Tabs item 2</bolt-text>
//           <bolt-text>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</bolt-text>
//         </bolt-tab-panel>
//         <bolt-tab-panel>
//           <bolt-text slot="label">Tabs item 3</bolt-text>
//           <bolt-text>Lorem ipsum dolor sit, amet consectetur adipisicing elit.</bolt-text>
//         </bolt-tab-panel>`;

//       document.body.appendChild(tabs);
//       tabs.useShadow = false;
//       tabs.updated();

//       const child = tabs.querySelector('bolt-tab-panel');
//       child.updated();

//       return child.renderRoot.innerHTML;
//     });

//     const renderedShadowDomHTML = await html(defaultTabsShadowRoot);

//     await page.waitFor(500);
//     const image = await page.screenshot();

//     expect(image).toMatchImageSnapshot({
//       failureThreshold: '0.01',
//       failureThresholdType: 'percent',
//     });

//     expect(renderedShadowDomHTML).toMatchSnapshot();
//   });
// });
