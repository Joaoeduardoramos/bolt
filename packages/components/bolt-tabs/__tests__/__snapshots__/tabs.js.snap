// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Tabs Align: center 2`] = `
<bolt-tabs
  align="center"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Align: end 2`] = `
<bolt-tabs
  align="end"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Align: start 2`] = `
<bolt-tabs
  align="start"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Inset: auto 2`] = `
<bolt-tabs
  inset="auto"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Inset: off 2`] = `
<bolt-tabs
  inset="off"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Inset: on 2`] = `
<bolt-tabs
  inset="on"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Twig usage 1`] = `
<bolt-tabs>
  <replace-with-grandchildren class="c-bolt-tabs c-bolt-tabs--align-start c-bolt-tabs--inset">
    <input type="radio"
           name="tabs-12345"
           id="tabs-12345-1"
           class="c-bolt-tabs__input"
           checked
    >
    <label for="tabs-12345-1"
           class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
    >
      <span class="c-bolt-tabs__label-inner">
        <span class="c-bolt-tabs__label-text">
          Tab label 1
        </span>
      </span>
    </label>
    <replace-with-children class="c-bolt-tab-panel__outer">
      <bolt-tab-panel ssr-hydrate="keep">
        <div slot="label">
          Tab label 1
        </div>
        <replace-with-grandchildren class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small">
          <div role="tabpanel"
               class="c-bolt-tab-panel__content"
          >
            This is the tab content.
          </div>
        </replace-with-grandchildren>
      </bolt-tab-panel>
    </replace-with-children>
    <input type="radio"
           name="tabs-12345"
           id="tabs-12345-2"
           class="c-bolt-tabs__input"
    >
    <label for="tabs-12345-2"
           class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
    >
      <span class="c-bolt-tabs__label-inner">
        <span class="c-bolt-tabs__label-text">
          Tab label 2
        </span>
      </span>
    </label>
    <replace-with-children class="c-bolt-tab-panel__outer">
      <bolt-tab-panel ssr-hydrate="keep">
        <div slot="label">
          Tab label 2
        </div>
        <replace-with-grandchildren class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small">
          <div role="tabpanel"
               class="c-bolt-tab-panel__content"
          >
            This is the tab content.
          </div>
        </replace-with-grandchildren>
      </bolt-tab-panel>
    </replace-with-children>
    <input type="radio"
           name="tabs-12345"
           id="tabs-12345-3"
           class="c-bolt-tabs__input"
    >
    <label for="tabs-12345-3"
           class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
    >
      <span class="c-bolt-tabs__label-inner">
        <span class="c-bolt-tabs__label-text">
          Tab label 3
        </span>
      </span>
    </label>
    <replace-with-children class="c-bolt-tab-panel__outer">
      <bolt-tab-panel ssr-hydrate="keep">
        <div slot="label">
          Tab label 3
        </div>
        <replace-with-grandchildren class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small">
          <div role="tabpanel"
               class="c-bolt-tab-panel__content"
          >
            This is the tab content.
          </div>
        </replace-with-grandchildren>
      </bolt-tab-panel>
    </replace-with-children>
  </replace-with-grandchildren>
</bolt-tabs>
`;

exports[`Bolt Tabs Web Component usage (Light DOM) 2`] = `
<bolt-tabs
  no-shadow=""
  ready=""
  style=""
>
  <!---->
  
       
      
  <div
    class="c-bolt-tabs c-bolt-tabs--align-start c-bolt-tabs--inset"
    style=""
  >
    
        
    <div
      class="c-bolt-tabs__nav"
      role="tablist"
      style=""
    >
      
          
      <!---->
      
          
      <bolt-trigger
        aria-controls="tab-panel-12345-1"
        aria-selected="true"
        class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
        id="tab-12345-1"
        no-outline=""
        role="tab"
        style=""
        tabindex="0"
      >
        
            
        <span
          class="c-bolt-tabs__label-inner"
          style=""
        >
          <span
            class="c-bolt-tabs__label-text"
            style=""
          >
            <!---->
            Tab label 1
            <!---->
          </span>
        </span>
        
          
      </bolt-trigger>
      
        
      <!---->
      
          
      <bolt-trigger
        aria-controls="tab-panel-12345-2"
        aria-selected="false"
        class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
        id="tab-12345-2"
        no-outline=""
        role="tab"
        style=""
        tabindex="-1"
      >
        
            
        <span
          class="c-bolt-tabs__label-inner"
          style=""
        >
          <span
            class="c-bolt-tabs__label-text"
            style=""
          >
            <!---->
            Tab label 2
            <!---->
          </span>
        </span>
        
          
      </bolt-trigger>
      
        
      <!---->
      
          
      <bolt-trigger
        aria-controls="tab-panel-12345-3"
        aria-selected="false"
        class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
        id="tab-12345-3"
        no-outline=""
        role="tab"
        style=""
        tabindex="-1"
      >
        
            
        <span
          class="c-bolt-tabs__label-inner"
          style=""
        >
          <span
            class="c-bolt-tabs__label-text"
            style=""
          >
            <!---->
            Tab label 3
            <!---->
          </span>
        </span>
        
          
      </bolt-trigger>
      
        
      <!---->
      
        
    </div>
    
        
    <div
      class="c-bolt-tabs__panels-container"
      style=""
    >
      
          
      <!---->
      <!---->
      
    
      <!---->
      <bolt-tab-panel
        no-shadow=""
        selected=""
        style=""
      >
        <!---->
        
       
      
        <div
          class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small"
          style=""
        >
          <!---->
          <!---->
          <!---->
          <!---->
          <div
            slot="label"
            style=""
          >
            Tab label 1
          </div>
          <!---->
          <!---->
          <!---->
          
                
          <div
            aria-expanded="true"
            aria-labelledby="tab-label-12345-1"
            class="c-bolt-tab-panel__content"
            id="tab-panel-12345-1"
            role="tabpanel"
            style=""
            tabindex="0"
          >
            
                  
            <!---->
            <!---->
            
      
            <!---->
            
      Tab panel 1
    
            <!---->
            <!---->
            
                
          </div>
          
              
          <!---->
          <!---->
        </div>
        
    
    
        <!---->
      </bolt-tab-panel>
      <!---->
      
    
      <!---->
      <bolt-tab-panel
        no-shadow=""
        style=""
      >
        <!---->
        
       
      
        <div
          class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small"
          style=""
        >
          <!---->
          <!---->
          <!---->
          <!---->
          <div
            slot="label"
            style=""
          >
            Tab label 2
          </div>
          <!---->
          <!---->
          <!---->
          
                
          <div
            aria-expanded="false"
            aria-labelledby="tab-label-12345-2"
            class="c-bolt-tab-panel__content"
            id="tab-panel-12345-2"
            role="tabpanel"
            style=""
            tabindex="0"
          >
            
                  
            <!---->
            <!---->
            
      
            <!---->
            
      Tab panel 2
    
            <!---->
            <!---->
            
                
          </div>
          
              
          <!---->
          <!---->
        </div>
        
    
    
        <!---->
      </bolt-tab-panel>
      <!---->
      
    
      <!---->
      <bolt-tab-panel
        no-shadow=""
        style=""
      >
        <!---->
        
       
      
        <div
          class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small"
          style=""
        >
          <!---->
          <!---->
          <!---->
          <!---->
          <div
            slot="label"
            style=""
          >
            Tab label 3
          </div>
          <!---->
          <!---->
          <!---->
          
                
          <div
            aria-expanded="false"
            aria-labelledby="tab-label-12345-3"
            class="c-bolt-tab-panel__content"
            id="tab-panel-12345-3"
            role="tabpanel"
            style=""
            tabindex="0"
          >
            
                  
            <!---->
            <!---->
            
      
            <!---->
            
      Tab panel 3
    
            <!---->
            <!---->
            
                
          </div>
          
              
          <!---->
          <!---->
        </div>
        
    
    
        <!---->
      </bolt-tab-panel>
      <!---->
      
  
      <!---->
      <!---->
      
        
    </div>
    
      
  </div>
  
    
    
  <!---->
</bolt-tabs>
`;

exports[`Bolt Tabs Web Component usage (Shadow DOM) 2`] = `
<bolt-tabs
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;
